router/beforeEach.ts

```diff
@@ -15,6 +16,8 @@ export function beforeEach (to: Route, from: Route, next) {
   if (multistoreEnabled) {
     if (storeCode in stores && to.name === storeCode + '-checkout') {
       window.location.replace(stores[storeCode].cmsUrl + '/vue/cart/sync/token/' + userToken + '/cart/' + cartToken)
+    } else if (to.name === 'checkout') {
+      window.location.replace(stores[config.defaultStoreCode].cmsUrl + '/vue/cart/sync/token/' + userToken + '/cart/' + cartToken)
```

Comment:
```typescript
} else if (storeCode in stores && to.name === 'checkout'  && stores[storeCode].cmsUrl !== undefined) {
  window.location.replace(stores[storeCode].cmsUrl + '/vue/cart/sync/token/' + userToken + '/cart/' + cartToken)
}
```

Isn't that the right code we would need here ?

Reply:
In my case your if condition still wouldn't work because of the `to.name === 'checkout'` but I guess you could use `currentStoreView().storeCode`.

Reply:
That i don't really understand? 

You are doing the `to.name === 'checkout'` too or are you meaning something else that would not work for you?

Reply:
Sorry, you're right. 

I meant the `if (storeCode in stores` part. This doesn't work because of  `const storeCode = storeCodeFromRoute(to)`.
In my case there is no storeCode in the url of the main store.

Reply:
That is why the `const storeCode = storeCodeFromRoute(to)` should be replaced with `const storeCode = currentStoreView().storeCode` then you have your storeCode.

Reply:
I see it now. I'm gonna try to commit this in my PR branch.

Reply:
I've commited your solution to the branch.

Reply:
Tanks for the fix.

Now we only need to wait until one finds the time to merge this.